### DISCUSSION TITLE
Building meta array in Lua to iterate in template

### DISCUSSION THREAD

**Discussion by lifeunleaded** [jgm/pandoc#10254] [Q&A] [answered]
I'm using an adaptation of @tarleb 's [`abstract-section`](https://github.com/pandoc-ext/abstract-section/blob/main/_extensions/abstract-section/abstract-section.lua) to build a table (array) of sections in `Meta` in a Lua filter, to iterate over in a template. Below is part of the filter that puts a section inside a nested array in `Meta` named `sections`. The reason is that I'd like to iterate over `sections` in the template.

```
  meta['sections'] = {}
  for metakey in pairs(section_identifiers) do
    metakey = stringify(metakey)
    local section = collected[metakey]
    if not meta[metakey] and section and #section > 0 then
      meta['sections'][metakey] = pandoc.MetaBlocks(section)
    end
  end
```

When I try to access it in the template, I can access things in `sections` directly by knowing their index (the `identifier`), but not iterate over them with `$for$`.

A test template with these (using docbook output to test):

```
<remark>sections loop</remark>
$for(sections)$
  $sections$
$endfor$
<remark>sections</remark>
$sections$
<remark>sections.myheading</remark>
$sections.myheading$
```

Gives me this:
```
<remark>sections loop</remark>
  true
<remark>sections</remark>
true
<remark>sections.myheading</remark>
[correct expected content for myheading, an item in sections] 
```

How should I form the `sections` table in Lua to be able to iterate over it in the template?

**Comment by lifeunleaded** [comment, marked as answer]
Having written the question, I realized the answer: index the table as a 1-indexed array, not with arbitrary identifiers.

```
meta['sections'] = {}
  i = 1
  for metakey in pairs(section_identifiers) do
    print('metakey', metakey)
    metakey = stringify(metakey)
    local section = collected[metakey]
    if section and #section > 0 then
      meta['sections'][i] = pandoc.MetaBlocks(section)
      i = i + 1
    end
  end
```

This gives me correct results when using `$for$` in the template.

**Reply by tarleb** [reply]
Alternatively, using `sections/pairs` should work, too. See https://pandoc.org/MANUAL#pipes.
